# Asymptotically Exact Confidence Intervals

Recall from class that if $\hat\theta$ is the maximum likelihood estimator for some parameter $\theta$, then $\hat\theta\pm\dfrac{z_{\alpha/2}}{\sqrt{nI(\hat\theta)}}$ is an asymptotically exact $100(1-\alpha)\%$ confidence interval for $\theta$.

In class, we saw that if $X_1,\ldots,X_n\sim\operatorname{Exp(\lambda)}$, we used this to construct an asymptotically exact confidence interval for the rate parameter $\lambda$.  In particular, $\dfrac{1}{\overline X}\pm \dfrac{z_{\alpha/2}}{\sqrt{n}\cdot\overline X}$.

Let's construct such an interval with $n = 120$ and $1-\alpha = 0.98$ (and so $\frac{\alpha}{2}=0.01$).  We'll use $\lambda = 4$ and check to see if the interval contains it.

In [3]:
x <- rexp(120, 4)
z <- qnorm(0.01, lower.tail = FALSE)
1/mean(x) - z/(sqrt(120) * mean(x))
1/mean(x) + z/(sqrt(120) * mean(x))

[1] 3.609749

[1] 5.556303

Indeed, $3.6097 < 4 < 5.5563$.

Now, let's check with a Monte Carlo simulation.  We would expect that approximately 98% of the time, the interval that we construct would contain 4.

In [18]:
z <- qnorm(0.01, lower.tail = FALSE)
mean(replicate(10000, {
    x <- rexp(120, 4);
    midpoint <- 1/mean(x);
    E <- z/(sqrt(120) * mean(x));
    midpoint - E < 4 && 4 < midpoint + 3}))

[1] 0.9928